test/unit_test/rag/test_find_codec.py

```
@@ -0,0 +1,153 @@
+    # Regression: the repo-pinned chardet 5.2.0 reports Windows-1254 at 0.56
+    # confidence, so trusting it before checking UTF-8 mis-decodes this blob.
+    text = "Sales are up 20% this quarter \U0001f4c8\U0001f680"
+    blob = text.encode("utf-8")
+    assert blob.decode(find_codec(blob)) == text
+
+
+@pytest.mark.p2
+@pytest.mark.parametrize(
+    "text, encoding",
+    [
+        ("Ежедневный отчет о продажах за прошлую неделю показывает рост.", "cp1251"),
```

Ruff (0.16.2)

[warning] String contains ambiguous `о` (CYRILLIC SMALL LETTER O). Did you mean `o` (LATIN SMALL LETTER O)?

(RUF001)